File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        int n=intervals.length;
        List<int[]> result = new ArrayList<>();
        
        //Sort based on start time of the intervals
        Arrays.sort(intervals, (a,b) -> a[0]-b[0]);
        
        for(int i=0;i<n;i++){
            if(result.isEmpty() || result.get(result.size()-1)[1] < intervals[i][0]){
                result.add(intervals[i]);
            }
            else{
                result.get(result.size() - 1)[1] = Math.max(result.get(result.size() - 1)[1], intervals[i][1]);
            }
        }
        return result.toArray(new int[result.size()][]);
    }
}

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    static bool compare(vector<int>& a, vector<int>& b){
        return a[0]<b[0];
    }
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> result;
        //Sort based on start time of the intervals
        sort(intervals.begin(), intervals.end(), compare);
        int n=intervals.size();
        for(int i=0;i<n;i++){
            if(result.empty() || result[result.size()-1][1] < intervals[i][0]){
                result.push_back(intervals[i]);
            }
            else{
                result[result.size()-1][1] = max(result[result.size()-1][1], intervals[i][1]);
            }
        }
        return result;
    }
};